library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphFeaturesTest.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.tinkerpop;
+
+import org.apache.tinkerpop.gremlin.structure.Graph.Features;
+
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphEdgeFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphGraphFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphVertexFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphVertexPropertyFeatures;
+
+public class GafferPopGraphFeaturesTest {
+    final GafferPopGraph graph = mock(GafferPopGraph.class);
+    final Features features = new GafferPopGraphFeatures();
+
+    @Test
+    public void shouldHaveCertainGafferPopGraphGraphFeatures() {
+        given(graph.features()).willReturn(features);
+        final Features graphFeatures = graph.features();
+
+        assertThat(graphFeatures.graph()).isInstanceOf(GafferPopGraphGraphFeatures.class);
+        assertThat(graphFeatures.graph().supportsTransactions()).isFalse();
+        assertThat(graphFeatures.graph().supportsThreadedTransactions()).isFalse();
+        assertThat(graphFeatures.graph().supportsComputer()).isFalse();
+    }
+
+    @Test
+    public void shouldHaveCertainGafferPopGraphVertexFeatures() {
+        final Object id = "testId";
+        given(graph.features()).willReturn(features);
+
+        final Features vertexFeatures = graph.features();
+
+        assertThat(vertexFeatures.vertex()).isInstanceOf(GafferPopGraphVertexFeatures.class);
+        assertThat(vertexFeatures.vertex().supportsRemoveVertices()).isFalse();
+        assertThat(vertexFeatures.vertex().supportsRemoveProperty()).isFalse();
+        assertThat(vertexFeatures.vertex().willAllowId(id)).isTrue();
+    }
+
+    @Test
+    public void shouldHaveCertainGafferPopGraphEdgeFeatures() {
+        final Object id = "testId";
+        given(graph.features()).willReturn(features);
+
+        final Features edgeFeatures = graph.features();
+
+        assertThat(edgeFeatures.edge()).isInstanceOf(GafferPopGraphEdgeFeatures.class);
+        assertThat(edgeFeatures.edge().supportsRemoveEdges()).isFalse();
+        assertThat(edgeFeatures.edge().supportsRemoveProperty()).isFalse();
+        assertThat(edgeFeatures.edge().willAllowId(id)).isTrue();
+    }
+
+    @Test
+    public void shouldReturnStringOfFeatures() {
+        given(graph.features()).willReturn(features);
+
+        assertThat(graph.features().toString()).contains("FEATURES", "ServiceCall");
```

Comment:
I think this contains a lot more than just what you are checking, maybe check for the headers? e.g. `EdgeFeatures` 

Full list as follows:

<details>

```text
FEATURES
> GraphFeatures
>-- Computer: false
>-- Transactions: false
>-- ThreadedTransactions: false
>-- Persistence: true
>-- ConcurrentAccess: true
>-- IoRead: true
>-- IoWrite: true
>-- OrderabilitySemantics: true
>-- ServiceCall: false
> VariableFeatures
>-- Variables: true
>-- DoubleArrayValues: true
>-- FloatArrayValues: true
>-- IntegerArrayValues: true
>-- StringArrayValues: true
>-- LongArrayValues: true
>-- SerializableValues: true
>-- StringValues: true
>-- UniformListValues: true
>-- BooleanValues: true
>-- ByteValues: true
>-- DoubleValues: true
>-- FloatValues: true
>-- IntegerValues: true
>-- LongValues: true
>-- MapValues: true
>-- MixedListValues: true
>-- BooleanArrayValues: true
>-- ByteArrayValues: true
> VertexFeatures
>-- RemoveVertices: false
>-- AddVertices: true
>-- MultiProperties: true
>-- DuplicateMultiProperties: true
>-- MetaProperties: true
>-- Upsert: false
>-- NullPropertyValues: true
>-- RemoveProperty: false
>-- AddProperty: true
>-- UserSuppliedIds: true
>-- NumericIds: true
>-- StringIds: true
>-- UuidIds: true
>-- CustomIds: true
>-- AnyIds: true
> VertexPropertyFeatures
>-- NullPropertyValues: true
>-- RemoveProperty: false
>-- UserSuppliedIds: true
>-- NumericIds: true
>-- StringIds: true
>-- UuidIds: true
>-- CustomIds: true
>-- AnyIds: true
>-- Properties: true
>-- DoubleArrayValues: true
>-- FloatArrayValues: true
>-- IntegerArrayValues: true
>-- StringArrayValues: true
>-- LongArrayValues: true
>-- SerializableValues: true
>-- StringValues: true
>-- UniformListValues: true
>-- BooleanValues: true
>-- ByteValues: true
>-- DoubleValues: true
>-- FloatValues: true
>-- IntegerValues: true
>-- LongValues: true
>-- MapValues: true
>-- MixedListValues: true
>-- BooleanArrayValues: true
>-- ByteArrayValues: true
> EdgeFeatures
>-- RemoveEdges: false
>-- Upsert: false
>-- AddEdges: true
>-- NullPropertyValues: true
>-- RemoveProperty: false
>-- AddProperty: true
>-- UserSuppliedIds: true
>-- NumericIds: true
>-- StringIds: true
>-- UuidIds: true
>-- CustomIds: true
>-- AnyIds: true
> EdgePropertyFeatures
>-- Properties: true
>-- DoubleArrayValues: true
>-- FloatArrayValues: true
>-- IntegerArrayValues: true
>-- StringArrayValues: true
>-- LongArrayValues: true
>-- SerializableValues: true
>-- StringValues: true
>-- UniformListValues: true
>-- BooleanValues: true
>-- ByteValues: true
>-- DoubleValues: true
>-- FloatValues: true
>-- IntegerValues: true
>-- LongValues: true
>-- MapValues: true
>-- MixedListValues: true
>-- BooleanArrayValues: true
>-- ByteArrayValues: true
```

</details>